### ISSUE TITLE
Change Interface to covariant IError<out TValue>

### ISSUE THREAD

**Issue by amyboose** [amantinband/error-or#52]
It would be great to have covariant interface instead of non-generic like this:

```C#
public interface IErrorOr<out TValue>
{
    List<Error>? Errors { get; }
    bool IsError { get; }
    List<Error> ErrorsOrEmptyList { get; }
    Error FirstError { get; }
    TValue Value { get; }
}
```

Then it allows us to upcast ErrorOr objects like this:

```C#
ErrorOr<TChild> childResult = new TChild();
IErrorOr<TParent> parentResult = childResult;
```

**Comment by amyboose** [comment]
Also it's possible to create additional interface 
```C#
public interface IErrorOr<out TValue> : IErrorOr
{
    TValue Value { get; }
}
```

**Comment by amantinband** [comment]
Good idea.

Implemented in #76

**Comment by amantinband** [comment]
should be available in version 1.5.0
